controller/control.go

```diff
@@ -443,32 +448,49 @@ func (c *Controller) RemoveReplicaNoLock(address string) error {
 					return fmt.Errorf("Cannot remove last replica if volume is up")
 				}
 			}
+			foundregrep = 0
 			for regrep := range c.RegisteredReplicas {
-				if strings.Contains(address, regrep) {
+				logrus.Infof("RemoveReplica ToRemove: %v Found: %v", address, regrep)
+				if address == "tcp://"+regrep+":9502" {
 					delete(c.RegisteredReplicas, regrep)
+					foundregrep = 1
+					break
 				}
 			}
+			if foundregrep == 0 {
+				logrus.Infof("RemoveReplica %v not found in registered replicas", address)
+			}
```

Comment:
should break from here

Reply:
Break is not required as all replicas will not go through the registration process , hence they will not be present in registered replica list. If the volume is already running, then atleast one of the replicas will be in R/W state. Therefore, the registration process to compare the quorum can be avoided.